src/formats/notion-api/database-helpers.ts

```diff
@@ -0,0 +1,864 @@
+			
+		case 'relation':
+			// Relation properties will be stored as list of links in page YAML
+			// Skip adding to .base file properties (will be handled in page frontmatter)
+			// But we still need to record it for reference
+			mappings[sanitizePropertyKey(key)] = {
+				displayName: propName,
+				type: OBSIDIAN_PROPERTY_TYPES.LIST,
+				isRelation: true,
+				relationConfig: prop.relation,
+			};
+			break;
+		
+		case 'rollup':
+			// Rollup properties should be converted to formulas in .base file
+			const rollupFormula = convertRollupToFormula(key, prop.rollup, notionProperties);
+			if (rollupFormula) {
+				mappings[`formula.${sanitizePropertyKey(key)}`] = {
+					displayName: propName,
+					formula: rollupFormula,
+				};
+			} else {
+				// Fallback: if conversion fails, log warning and skip this property
+				// Don't add it as a regular property since it should be a formula
+				console.warn(`Failed to convert rollup property "${propName}" to formula.`);
+			}
+			break;
+		
+		case 'people':
+		case 'files':
+		case 'created_time':
+		case 'created_by':
+		case 'last_edited_time':
+		case 'last_edited_by':
+			// These will be converted to text representation
+			mappings[sanitizePropertyKey(key)] = {
+				displayName: propName,
+				type: OBSIDIAN_PROPERTY_TYPES.TEXT,
+			};
+			break;
+			
+			default:
+				// Unsupported types -> text
+				console.log(`Unsupported property type: ${propType}, treating as text`);
+				mappings[sanitizePropertyKey(key)] = {
+					displayName: propName,
+					type: OBSIDIAN_PROPERTY_TYPES.TEXT,
+				};
+		}
+	}
+	
+	// Second pass: convert to ordered array
+	// If propertyIds is provided, use that order; otherwise use the order from Object.entries
+	const orderedMappings: Array<{key: string, config: any}> = [];
+	
+	if (propertyIds.length > 0) {
+		// Use the order from propertyIds
+		// Note: propertyIds contains the original Notion property IDs, so we need to sanitize them
+		for (const propId of propertyIds) {
+			const sanitizedKey = sanitizePropertyKey(propId);
+			if (mappings[sanitizedKey]) {
+				orderedMappings.push({
+					key: sanitizedKey,
+					config: mappings[sanitizedKey]
+				});
+			}
+			// Also check for formula properties (with "formula." prefix)
+			const formulaKey = `formula.${sanitizedKey}`;
+			if (mappings[formulaKey]) {
+				orderedMappings.push({
+					key: formulaKey,
+					config: mappings[formulaKey]
+				});
+			}
+		}
+		// Add any properties that weren't in propertyIds (shouldn't happen, but just in case)
+		for (const [key, config] of Object.entries(mappings)) {
+			const baseKey = key.replace(/^formula\./, '');
+			if (!propertyIds.map(id => sanitizePropertyKey(id)).includes(baseKey)) {
+				orderedMappings.push({ key, config });
+			}
+		}
+	} else {
+		// No propertyIds provided, use the order from Object.entries
+		for (const [key, config] of Object.entries(mappings)) {
+			orderedMappings.push({ key, config });
+		}
+	}
+	
+	return orderedMappings;
```

Comment:
Return and object with two lists instead of combining here and then separating again.